File: java/src/sample/replicant/sample3/MockWidgetRepository.java
// Copyright 2009 Kiel Hodges
package replicant.sample3;

import replicant.*;
import static replicant.Replicant.*;

public class MockWidgetRepository implements WidgetRepository {

  
  public MockWidgetRepository(Expecter expecter) {
    findWidget = methodMocker(this, "findWidget").
      mappingResponses().
      enforcingExpectationsWith(expecter).
      with1Arg();
    storeWidget = voidMethodMocker(this, "storeWidget").
      enforcingExpectationsWith(expecter).
      mappingResponses().
      with1Arg();
  }

  public Widget findWidget(int widgetId) {
    return findWidget.call(widgetId);
  }
  
  public final MethodMocker1<Widget, Integer> findWidget;
  
  
  
  public void storeWidget(Widget widget) {
    storeWidget.call(widget);
  }

  public final MethodMocker1<Void, Widget> storeWidget; 
  
}

File: scala/src/test/scala/sample2/MockWidgetRepository.scala
// Copyright 2011 Kiel Hodges
package sample2

import replicant._
import experiment1._

class MockWidgetRepository extends MockGenericRepository[Widget] { self =>
  
  override val testDouble: WidgetRepository = new TestDouble with WidgetRepository {
    def findByPartNumber(partNumber: String): Widget = self.findByPartNumber(partNumber).response
  }

  val findByPartNumber = method("findByPartNumber", testDouble.findByPartNumber _)
  
}
